crates/sqlexec/src/planner/logical_plan/delete.rs

```diff
@@ -0,0 +1,62 @@
+use datafusion::prelude::SessionContext;
+use protogen::metastore::types::catalog::TableEntry;
+
+use super::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Delete {
+    pub table: TableEntry,
+    pub where_expr: Option<Expr>,
+}
+
+impl UserDefinedLogicalNodeCore for Delete {
+    fn name(&self) -> &str {
+        Self::EXTENSION_NAME
+    }
+
+    fn inputs(&self) -> Vec<&DfLogicalPlan> {
+        Vec::new()
+    }
+
+    fn schema(&self) -> &datafusion::common::DFSchemaRef {
+        // TODO: Use correct schema
+        &EMPTY_SCHEMA
+    }
+
+    fn expressions(&self) -> Vec<datafusion::prelude::Expr> {
+        Vec::new()
+    }
+
+    fn fmt_for_explain(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", Self::EXTENSION_NAME)
+    }
+
+    fn from_template(
+        &self,
+        _exprs: &[datafusion::prelude::Expr],
+        _inputs: &[DfLogicalPlan],
+    ) -> Self {
+        self.clone()
+    }
+}
+
+impl ExtensionNode for Delete {
+    type ProtoRepr = protogen::sqlexec::logical_plan::Delete;
+    const EXTENSION_NAME: &'static str = "Delete";
+
+    fn try_decode(
+        _proto: Self::ProtoRepr,
+        _ctx: &SessionContext,
+        _codec: &dyn LogicalExtensionCodec,
+    ) -> std::result::Result<Self, ProtoConvError> {
+        unimplemented!()
+    }
+
+    fn try_decode_extension(_extension: &LogicalPlanExtension) -> Result<Self> {
+        unimplemented!()
+    }
+
+    fn try_encode(&self, _buf: &mut Vec<u8>, _codec: &dyn LogicalExtensionCodec) -> Result<()> {
+        unimplemented!()
+    }
+}
```

Comment:
don't we need this implemented for (de)serialization of the node?

Reply:
I thought we don't need to serialise and deserialise logical plan anymore?

Reply:
I think we'll likely be removing some (or all?) of this trait since we only care able serializing physical plans now.